xiaomi_gateway/__init__.py

```diff
@@ -95,6 +94,7 @@ def discover_gateways(self):
                 for gateway in self._gateways_config:
                     sid = gateway.get('sid')
                     if sid is None or sid == resp["sid"]:
```

Comment:
There's still some buggy code here. I know it should never happen, because in HA docs it is written that you have to provide mac when using more than 1 gateway, but IF the sid is None, then we're just setting discovery_retries and gateway_key over and over again for every gateway.
Shouldn't we like set the sid to thee value from the response, when it is None?
Or even crash if _gateways_config size is bigger than 1 and no sid is set.

Reply:
And shouldn't we check in line 109 if the key with the given IP already exists not to create XiaomiGateway object two times?

Reply:
I will provide some details soon. You need two know some additional details. ;-)

Reply:
Can't wait ;)

Reply:
Still waiting... ;)